data/teca_coordinate_util.cxx

```diff
@@ -59,6 +62,45 @@ int time_step_of(p_teca_double_array time, bool lower,
     return -1;
 #endif
 }
+    
+    
+// given a time value (val), associated time units (units), and calendar (calendar), return a human-readable
+// rendering of the date (date) in a strftime-format (format).  return 0 if successful.
+int time_step_to_string(double val, 
+                        const std::string &calendar,  const std::string &units, const std::string &format,
+                        std::string &date)
+{
+#if defined(TECA_HAS_UDUNITS)
+    
+    // use calcalcs to convert val to a set of year/month/day/etc.
+    struct tm timedata;
+    double seconds = -1;
+    if ( calcalcs::date(val, &timedata.tm_year, &timedata.tm_mon, &timedata.tm_mday, &timedata.tm_hour,
+            &timedata.tm_min, &seconds, units.c_str(), calendar.c_str()))
+    {
+        TECA_ERROR("failed to convert relative time value \"" << val
+            << "\" to with the calendar \"" << calendar
+            << "\" and units \"" << units << "\".")
+        return -1;
+    }
+    
+    // fix some of the timedata info to conform to what strftime expects
+    timedata.tm_year -= 1900; // make the year relative to 1900
+    timedata.tm_mon -= 1; // make the months start at 0 instead of 1
+    timedata.tm_sec = (int) std::round(seconds);
+    
+    // use std::put_time to convert the time data to a string
+    std::ostringstream os;
```

Comment:
use strftime, given that we don't need the ostream (as discussed I'll make that change before we merge)